core/src/main/scala/shapeless/lazy.scala

```diff
@@ -164,9 +164,34 @@ object Strict {
 }
 
 @macrocompat.bundle
-class LazyMacros(val c: whitebox.Context) extends CaseClassMacros {
+trait OpenImplicitMacros extends compat.CompatLite {
+  val c: whitebox.Context
+
+  import c.universe._
+
+  def openImplicitTpe: Option[Type] =
+    c.openImplicits.headOption.map(x => (x: ImplicitCandidate211).pt)
```

Comment:
Yes, I'd like to avoid reintroducing any non-test variant source here. If there is a problem it would need to be fixed in macro-compat.